File: src/evai/command/JsCommands.java
package evai.command;

import mindustry.*;
import mindustry.logic.*;
import java.util.*;

public class JsCommands implements CommandRegistry.CommandHandler {
 @Override
 public CommandResult handle(String action, String[] args) {
 switch (action) {
 case "js": return js(args);
 case "logic": return logic(args);
 case "eval": return eval(args);
 default: return new CommandResult(false, "unknown js command: " + action);
 }
 }

 private CommandResult js(String[] args) {
 if (args.length == 0) return new CommandResult(false, "usage: js <code>");
 String code = String.join(" ", args);
 try {
 Object result = Vars.mods.getScripts().runConsole(code);
 Map<String, Object> data = new HashMap<>();
 data.put("code", code);
 data.put("result", result != null ? result.toString() : "undefined");
 return new CommandResult(true, "", data);
 } catch (Exception e) {
 return new CommandResult(false, "JS error: " + e.getMessage());
 }
 }

 private CommandResult logic(String[] args) {
 if (args.length == 0) return new CommandResult(false, "usage: logic <code>");
 String code = String.join(" ", args);
 try {
                LExecutor.runLogicScript(code);
                Map<String, Object> data = new HashMap<>();
                data.put("code", code);
                data.put("result", "executed");
                return new CommandResult(true, "", data);
 } catch (Exception e) {
 return new CommandResult(false, "Logic error: " + e.getMessage());
 }
 }

 private CommandResult eval(String[] args) {
 if (args.length == 0) return new CommandResult(false, "usage: eval <expr>");
 String expr = String.join(" ", args);
 try {
 Object result = Vars.mods.getScripts().runConsole("print(" + expr + ");");
 Map<String, Object> data = new HashMap<>();
 data.put("expression", expr);
 data.put("result", result != null ? result.toString() : "undefined");
 return new CommandResult(true, "", data);
 } catch (Exception e) {
 return new CommandResult(false, "Eval error: " + e.getMessage());
 }
 }
}

File: src/evai/command/JsCommands.kt
package evai.command

import mindustry.*
import evai.command.CommandRegistry.CommandHandler

class JsCommands : CommandHandler {
    override fun handle(action: String, args: Array<String>): CommandResult = when (action) {
        "js" -> {
            if (args.isEmpty()) return CommandResult(false, "usage: js <code>")
            val code = args.joinToString(" ")
            try {
                val result = Vars.mods.getScripts().runConsole(code)
                CommandResult(true, data = mapOf("code" to code, "result" to (result?.toString() ?: "undefined")))
            } catch (e: Exception) {
                CommandResult(false, "JS error: ${e.message}")
            }
        }
        "logic" -> {
            if (args.isEmpty()) return CommandResult(false, "usage: logic <code>")
            val code = args.joinToString(" ")
            try {
                val result = mindustry.logic.LExecutor.runLogicScript(code)
                CommandResult(true, data = mapOf("code" to code, "result" to result.toString()))
            } catch (e: Exception) {
                CommandResult(false, "Logic error: ${e.message}")
            }
        }
        "eval" -> {
            if (args.isEmpty()) return CommandResult(false, "usage: eval <expression>")
            val expr = args.joinToString(" ")
            try {
                val result = Vars.mods.getScripts().runConsole("print($expr);")
                CommandResult(true, data = mapOf("expression" to expr, "result" to (result?.toString() ?: "undefined")))
            } catch (e: Exception) {
                CommandResult(false, "Eval error: ${e.message}")
            }
        }
        else -> CommandResult(false, "unknown js command: $action")
    }
}